packages/commonwealth/server/passport/magic.ts

```diff
@@ -207,6 +206,59 @@ async function createNewMagicUser({
   });
 }
 
+// Replaces ghost addresses once replacements have been generated
+async function replaceGhostAddresses(
+  existingUserInstance: UserInstance,
+  addressInstances: AddressInstance[],
+  models: DB,
+  transaction: Transaction,
+) {
+  const ghostAddresses = (existingUserInstance?.Addresses?.filter(
+    ({ ghost_address }: AddressAttributes) => !!ghost_address,
+  ) || []) as AddressAttributes[];
+
+  for (const ghost of ghostAddresses) {
+    const replacementAddress = addressInstances.find(
```

Comment:
So we just pick an arbitrary address the user owns on the same community? This should be OK but maybe we should in the future surface a UI to allow them to decide which migration address to use.

Reply:
This is a copy/paste from below, just encapsulating the replacement in a function